agent/eventhandler/task_handler.go

```diff
@@ -114,10 +114,12 @@ func (handler *TaskHandler) flushBatch(event *api.TaskStateChange) {
 	delete(handler.tasksToContainerStates, event.TaskARN)
 }
 
-// Prepares a given event to be sent by adding it to the handler's appropriate
-// eventList
+// addEvent prepares a given event to be sent by adding it to the handler's appropriate
+// eventList and remove the entry in tasksToEvents map
 func (handler *TaskHandler) addEvent(change *sendableEvent, client api.ECSClient) {
 	seelog.Info("TaskHandler, Adding event: ", change)
```

Comment:
Can you move this below the `taskHandlerLock` statements? Also, you can change it to use `Infof` while you're here.